File: GoLang/solution/solution1282.go
package solution

func GroupThePeople(groupSizes []int) [][]int {
	res := make([][]int, 0)
	temp := make([][]int, len(groupSizes) + 1)
	for i := 0; i < len(groupSizes); i++ {
		if temp[groupSizes[i]] == nil || len(temp[groupSizes[i]]) == 0 {
			temp[groupSizes[i]] = make([]int, 0)
		}
		temp[groupSizes[i]] = append(temp[groupSizes[i]], i)
		if len(temp[groupSizes[i]]) == groupSizes[i] {
			res = append(res, temp[groupSizes[i]])
			temp[groupSizes[i]] = make([]int, 0)
		}
	}
	return res
}

File: java/src/Solution1282.java
import java.util.ArrayList;
import java.util.List;

public class Solution1282 {
	public List<List<Integer>> groupThePeople(int[] groupSizes) {
		List<List<Integer>> res = new ArrayList<>();
		List<List<Integer>> temp = new ArrayList<>();
		for (int i = 0; i < groupSizes.length + 1; i++) {
			temp.add(new ArrayList<>());
		}
		for (int i = 0; i < groupSizes.length; i++) {
			List<Integer> t = temp.get(groupSizes[i]);
			t.add(i);
			if (t.size() == groupSizes[i]) {
				List<Integer> r = new ArrayList<>(t);
				res.add(r);
				t.clear();
			}
		}
		return res;
	}
}
